[Build/CI Change] Publish to test pypi during the release workflow

hinxcode opened the pull request:
<!--
Thank you for your contribution to the repo :)

Pull Request (PR) Instructions:
Provide a general summary of your changes in the Title above. Fill out each section of the template, and replace the space with an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help! Once you are satisfied with the pull request, click the "Create pull request" button to submit it for review.

Before submitting this PR, please ensure that your input and responses are entered in the designated space provided below each section to keep all project-related information organized and easily accessible.

How to link to a PR:
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
-->

## Change Description
<!---
Describe your changes in detail. In your description, you should answer questions like "Why is this change required? What problem does it solve?".

If it fixes an open issue, please link to the issue here. If this PR closes an issue, put the word 'closes' before the issue link to auto-close the issue when the PR is merged.
-->
- [x] My PR includes a link to the issue that I am addressing

This PR is addressing issue tickets:
https://github.com/AutoResearch/autodoc/issues/24 
https://github.com/uw-ssec/codeuw/issues/47

## Solution Description
<!-- Please explain the technical solution that I have provided and how it addresses the issue or feature being implemented -->

**1. Publish to Test PyPI**
I separated actions for building the package from the `deploy` job, making a shareable artifact between publishing jobs to PyPI and Test PyPI.
**2. Check PyPI packages**
Similar to the testing done in NoisePy. The changes in __init__.py are for retrieving the version number.

## Code Quality
- [x] I have read the Contribution Guide
- [x] My code follows the code style of this project
- [x] My code builds (or compiles) cleanly without any errors or warnings
- [x] My code contains relevant comments and necessary documentation

I've tested these changes using [nektos/act](https://github.com/nektos/act) locally and my own forked [repo](https://github.com/hinxcode/autodoc/actions/runs/9922967698) without errors.

## Project-Specific Pull Request Checklists
<!--- Please only use the checklist that apply to your change type(s) -->

### Build/CI Change Checklist
- [ ] If required or optional dependencies have changed (including version numbers), I have updated the README to reflect this
- [ ] If this is a new CI setup, I have added the associated badge to the README

A new token `TEST_PYPI_API_TOKEN` is needed to add to the repo.

hinxcode committed:
feat: Add version number

hinxcode committed:
chore: Update publish action

hinxcode committed:
chore: Add job's name

carlosgjs commented on .github/workflows/publish-to-pypi.yml:
```diff
@@ -31,6 +30,74 @@ jobs:
+    - name: Publish distribution 📦 to TestPyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        password: ${{ secrets.TEST_PYPI_API_TOKEN }}
```
Can you add a `TODO:` comment that this secrets needs to be setup?

carlosgjs reviewed (approved):
Looks good, thank you for picking up this issue! I just added a couple of minor comments, can you address them?

hinxcode committed:
chore: Update comments and echos

hinxcode commented:
> Looks good, thank you for picking up this issue! I just added a couple of minor comments, can you address them?

@carlosgjs Hi, I'm glad to help! I've updated the PR. Thanks for your feedback!
